tests/integration/test_where.py

```diff
@@ -45,6 +45,98 @@ def test_condition(cond):
     assert np.array_equal(np.where(anp, xnp, ynp), num.where(a, x, y))
 
 
+@pytest.mark.parametrize(
+    "shape_a",
+    ((1,), (3,), (1, 3), (3, 3), (2, 3, 3)),
+    ids=lambda shape_a: f"(shape_a={shape_a})",
+)
+def test_broadcast(shape_a):
+    a = mk_seq_array(num, shape_a)
+    anp = mk_seq_array(np, shape_a)
+    cond = a > 5
+    cond_np = anp > 5
+
+    shape_x = (3, 3)
+    x = mk_seq_array(num, shape_x)
+    xnp = mk_seq_array(np, shape_x)
+    shape_y = (1, 3)
+    y = mk_seq_array(num, shape_y) * 10
+    ynp = mk_seq_array(np, shape_y) * 10
+
+    assert np.array_equal(np.where(cond_np, xnp, ynp), num.where(cond, x, y))
+
+
+@pytest.mark.xfail
+def test_condition_none():
+    # In Numpy, pass and returns [1, 2]
+    # In cuNumeric, raises AttributeError:
+    # 'NoneType' object has no attribute '_maybe_convert'
+    x = 0
+    ynp = np.array([1, 2])
+    y = num.array(ynp)
+    assert np.array_equal(np.where(None, x, ynp), num.where(None, x, y))
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize(
+    "values",
+    ((None, None), (None, 1), (1, None)),
+    ids=lambda values: f"(values={values})",
+)
+def test_x_y_none(values):
+    # For x=None and y=None,
+    # In Numpy, pass and returns [None, None]
+    # In cuNumeric, pass and returns (array([0]),)
+    # For x=None and y=1
+    # In Numpy, pass and returns [None, 1]
+    # In cuNumeric, raises ValueError: both 'x' and 'y' parameters
+    # must be specified together for where
+    cond = [True, False]
+    anp = np.array(cond)
+    a = num.array(anp)
+    x, y = values
+    assert np.array_equal(np.where(anp, x, y), num.where(a, x, y))
```

Comment:
mypy `reveal_type(np.where)` says:

```
Overload(
  def (ArrayLike) -> builtins.tuple[numpy.ndarray[Any, numpy.dtype[numpy.signedinteger[Any]]], ...],
  def (ArrayLike, ArrayLike, ArrayLike) -> numpy.ndarray[Any, numpy.dtype[Any]]
)
```

so again the behavior on `None` is undocumented.

I think this is a case where NumPy uses a special (non-None) "no input" value as the default argument value, whereas cuNumeric just uses `None`.

```
np.where(c, None, None) --(undocumented)-> np.where(c, np.ndarray(None, dtype=object), np.ndarray(None, dtype=object))
cn.where(c, None, None) -> cn.where(c) -> success

np.where(c, None, 1) -> np.where(c, np.ndarray(None, dtype=object), np.ndarray(1, dtype=int))
cn.where(c, None, 1) -> "missing array x"
```

Both approaches work for documented cases, so I don't really feel obliged to change our input handling to use a non-None "no input" value, just so we can copy NumPy's undocumented special handling of `None` (which we can't really do anything meaningful with anyway). But feel free to challenge me.